File: rb/features.rb
# Myip SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module MyipFeatures
  def self.make_feature(name)
    case name
    when "base"
      MyipBaseFeature.new
    when "test"
      MyipTestFeature.new
    else
      MyipBaseFeature.new
    end
  end
end

File: py/features.py
# Myip SDK feature factory

from feature.base_feature import MyipBaseFeature
from feature.test_feature import MyipTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MyipBaseFeature(),
        "test": lambda: MyipTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
